cmd-record.c

```diff
@@ -352,9 +354,46 @@ static int fill_file_header(struct opts *opts, int status, struct rusage *rusage
 	if (write(fd, &hdr, sizeof(hdr)) != (int)sizeof(hdr))
 		pr_err("writing header info failed");
 
+	char *orig_args = opts->args;
+	char *orig_retval = opts->retval;
+	if (opts->auto_args) {
+		/* add auto-args info to write into info file */
+		size_t alloc_size;
+
+		if (orig_args) {
+			alloc_size = strlen(auto_args_list) + strlen(orig_args) + 2;
+			opts->args = (char*)malloc(alloc_size);
```

Comment:
Please use xmalloc (and its friends) and you don't need to cast it in C.

Reply:
done!